bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,357 @@
+import org.openhab.binding.vdr.internal.svdrp.SVDRPDiskStatus;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPEpgEvent;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPException;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPParseResponseException;
+import org.openhab.binding.vdr.internal.svdrp.SVDRPVolume;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VDRHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Matthias Klocke - Initial contribution
+ */
+@NonNullByDefault
+public class VDRHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VDRHandler.class);
+
+    private VDRConfiguration config = new VDRConfiguration();
+
+    private @Nullable ScheduledFuture<?> refreshThreadFuture = null;
+
+    public VDRHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * when disposing refresh thread has to be cancelled
+     */
+    @Override
+    public void dispose() {
+        stopRefreshThread(true);
+    }
+
+    /**
+     * Initialization of {@link VDRHandler}
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(VDRConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // background initialization:
+        scheduler.execute(() -> {
+            stopRefreshThread(true);
+
+            SVDRPClient con = new SVDRPClientImpl(config.getHost(), config.getPort());
+            try {
+                con.openConnection();
+                updateStatus(ThingStatus.ONLINE);
+                updateProperties(con);
+                con.closeConnection();
+            } catch (SVDRPException se) {
+                logger.trace("VDR not online: Thing {}, Message: {}", this.getThing().getUID(), se.getMessage());
+                // also update power channel when thing is offline when initializing
+                thing.getChannels().stream().map(c -> c.getUID()).filter(this::isLinked).forEach(channelUID -> {
+                    try {
+                        if ("power".equals(channelUID.getIdWithoutGroup())) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
Removed try and catch as it was not necessary.